### PR TITLE
feat(TDI-50047): migrate to custombuilder rootless

### PR DESCRIPTION
**Pull request by acatoire**
https://jira.talendforge.org/browse/TDI-50047
Based on https://github.com/Talend/connectors-se/pull/1051
Add rootless in pod

### CODE REVIEW

File: .jenkins/jenkins_pod.yml [hunk 1]
```diff
@@ -2,34 +2,45 @@
...
       command: [ cat ]
       tty: true
       volumeMounts: [
-        { name: efs-jenkins-connectors-se-m2, mountPath: /root/.m2/repository },
-        { name: efs-jenkins-connectors-asdf, mountPath: /root/.asdf/installs, subPath: installs },
+        { name: efs-jenkins-connectors-se-m2-rootless,
+          mountPath: /home/jenkins/.m2/ },
+        { name: efs-jenkins-connectors-asdf-rootless,
+          mountPath: /home/jenkins/.asdf/installs,
+          subPath: installs },
```
Comment (quilicicf) [inline .jenkins/jenkins_pod.yml:21]: Didn't realize before but I think the subPath breaks the cache.
Please test it, I don't have one on my Jenkins instance and it doesn't work for TDOPS either (somehow, they had the same error, which I don't know the origin of).

File: Jenkinsfile [hunk 2]
```diff
@@ -297,6 +300,7 @@ pipeline {
                         """
                     }
 
+                    sleep(time:3, unit:"SECONDS") // Ensure the docker container is fully up and running.
```
Comment (quilicicf) [inline Jenkinsfile:303]: Sleep is a code smell (already commented in your previous PR, see more details there).

Comment (acatoire) [reply Jenkinsfile]: Retry implemented

File: .jenkins/jenkins_pod.yml (hunk 1 above)
Comment (acatoire) [reply .jenkins/jenkins_pod.yml]: Corrected

**Pull request closed without merging**